### ISSUE TITLE
ngclient: consider removing SLEEP_BEFORE_ROUND

### ISSUE THREAD

**Issue by sechkova** [theupdateframework/tuf#1484]
**Description of issue or feature request**:

As documented in the code SLEEP_BEFORE_ROUND exists because:
```
# The client's update procedure (contained within a while-loop) can potentially
# hog the CPU.  The following setting can be used to force the update sequence
# to suspend execution for a specified amount of time.  See
# theupdateframework/tuf/issue#338.
```
Consider whether this is still the case with the newer code and fix the issue if needed.
Remove the sleep constant.

**Comment by sechkova** [comment]
Reading #338 that led to adding the `sleep_before_round` config, it seems that we either keep it and close this issue or ignore the high CPU usage caused by the download while loop on embedded devices. 
Unless someone knows a better way to solve the problem I suggest we keep the config. @joshuagl @jku

**Comment by jku** [comment]
The issue I have with this "high cpu usage" problem is that it does not make a lot of sense and the actual issue was never debugged (or at least documented): A 1GHz arm7  should have absolutely no issues running simple download code -- if the `read()` call in the loop actually returns without blocking (for a timeout) and the code ends up looping billions of times then that was the real issue and the whole thing needed to be redesigned.

Luckily it looks like the download loop was redesigned at least once between 2016 and today -- we no longer use the `urllib.request.opener` API. So I'm totally fine with removing the sleep with a commit message stating that to our best knowledge current code will always block inside the loop: So the correct method of tweaking the compromise between CPU and download size granularity is to tweak `chunk_size` and `socket_timeout` (or whatever knobs the used Fetcher implementation decides to expose) but that we believe default values should work for pretty much any computer that can actually run python3.

TL;DR: 
* I don't believe in the high CPU usage theory
* If there is an issue, sleep is not a good solution
* pushing responsibility to user (in the form of configuration) is also not good

**Comment by sechkova** [comment]
I see, I'll test it with the current network stack and post the results.

**Comment by sechkova** [comment]
So I tried reproducing the #338 issue but it is quite impossible.
- There is no clear scenario in the discussion, only results of measured times
- Even if I tried to guess the scenario by the comments, I still could not reproduce the 2016 environment with significant efforts (quite unjustified)
- My hardware setup is probably quite different
- I tested with timing the current code (on a 4-cores VM so times mean nothing) and I see no signs of the issue or any improvements by tweaking the sleep before round constant.

```
time ~/tuf/scripts/client.py --repo http://localhost:8001 file1.txt

# SLEEP_BEFORE_ROUND = None
real	0m0.169s
user	0m0.118s
sys	0m0.045s

# SLEEP_BEFORE_ROUND = 0.05
real	0m0.518s
user	0m0.135s
sys	0m0.024s
```

As @jku had already explained, the download code as well as the networking libraries have gone through significant changes over the years. I am removing the constant for the ngclient.

**Comment by joshuagl** [comment]
Thanks for putting the effort in to investigate before removing

Completely agree with its removal 🔥
